jetbrains-core/src/software/aws/toolkits/jetbrains/core/plugins/ToolkitUpdateManager.kt

```diff
@@ -0,0 +1,103 @@
+import org.jetbrains.annotations.VisibleForTesting
+import software.aws.toolkits.core.utils.debug
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.AwsToolkit
+import software.aws.toolkits.jetbrains.settings.AwsSettings
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+
+class ToolkitUpdateManager {
+    private val alarm = Alarm(Alarm.ThreadToUse.SWING_THREAD)
+
+    init {
+        runTask()
+    }
+
+    @VisibleForTesting
+    internal fun runTask() {
+        if (alarm.isDisposed) return
+        scheduleUpdateTask()
+
+        if (!AwsSettings.getInstance().isAutoUpdateEnabled) return
+
+        runInEdt {
+            ProgressManager.getInstance().run(
+                object : Task.Backgroundable(null, "Updating AWS Toolkit") {
+                    override fun run(indicator: ProgressIndicator) {
+                        checkForUpdates(indicator)
+                    }
+                }
+            )
+        }
+    }
+
+    private fun scheduleUpdateTask() {
+        alarm.addRequest({ runTask() }, UPDATE_CHECK_INTERVAL_IN_MS)
+    }
+
+    @RequiresBackgroundThread
+    fun checkForUpdates(progressIndicator: ProgressIndicator) {
+        // Note: This will need to handle exceptions and ensure thread-safety
+        try {
+            val toolkitPlugin = PluginManagerCore.getPlugin(PluginId.findId(AwsToolkit.PLUGIN_ID)) ?: return
+            if (!toolkitPlugin.isEnabled) return
+            val latestToolkitPluginDownloader = getUpdate(toolkitPlugin) ?: return
+            if (!latestToolkitPluginDownloader.prepareToInstall(progressIndicator)) return
+            latestToolkitPluginDownloader.install()
+        } catch (e: Exception) {
+            LOG.debug(e) { "Unable to install AWS Toolkit" }
+            return
+        }
+        if (AwsSettings.getInstance().isAutoUpdateNotificationDoNotShowAgain) return
+        notifyInfo(
+            title = message("aws.notification.title"),
+            content = message("aws.settings.auto_update.notification.message"),
+            project = null,
+            notificationActions = listOf(
+                NotificationAction.createSimpleExpiring(message("aws.settings.auto_update.notification.yes")) {
+                    ApplicationManager.getApplication().restart()
+                },
+                NotificationAction.createSimpleExpiring(message("aws.settings.auto_update.notification.no")) {},
+                NotificationAction.createSimpleExpiring(message("aws.notification.do_not_show_again")) {
+                    AwsSettings.getInstance().isAutoUpdateNotificationDoNotShowAgain = true
+                }
+            )
+        )
+    }
+
+    @VisibleForTesting
+    internal fun getUpdate(pluginDescriptor: IdeaPluginDescriptor): PluginDownloader? =
+        getUpdateInfo().firstOrNull {
+            it.id == pluginDescriptor.pluginId && it.pluginVersion > pluginDescriptor.version
```

Comment:
```suggestion
            it.id == pluginDescriptor.pluginId && compareVersionsSkipBrokenAndIncompatible(it.pluginVersion, pluginDescriptor) > 0
```

string comparison for version numbers is not good enough